### PR TITLE
LPS-186875 Functional Automation: Create generic creation button actions

### PR DESCRIPTION
**Pull request by andreluis2022**
**Description:**
Create Automation tests LPS-197775 Create generic creation button actions

**JIRA Ticket:** 
https://liferay.atlassian.net/browse/LPS-186875

### CODE REVIEW

**Comment by andreluis2022** [comment]
ci:test:frontend-data-set

**Comment by andreluis2022** [comment]
ci:test:sf

**Comment by andreluis2022** [comment]
ci:test:frontend-data-set

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 1]
```diff
@@ -252,6 +252,40 @@ definition {
 		}
 	}
 
+	@description = "LPS-186875 Confirm that the Variant modal is displayed with the correct values"
+	@priority = 4
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:256]: Low this to 3 (Update jira ticket too)

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 2]
```diff
@@ -491,6 +525,53 @@ definition {
 		}
 	}
 
+	@description = "LPS-186875 Confirm that creation action changes are not saved by clicking the cancel button"
+	@priority = 4
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:529]: Low this to 3 (Update jira ticket too)

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 3]
```diff
@@ -2360,6 +2515,91 @@ definition {
...
+				value1 = "No actions were created.");
+
+			AssertTextPresent(
+				locator1 = "ObjectCustomViews#VIEW_BUILDER_EMPTY_STATE_TEXT",
+				value1 = "Start creating an action to interact with your data.");
+		}
+	}
+
+	@description = "LPS-186875 Confirm that an action creation can be saved with a different translation, and using the Side Panel type"
+	@priority = 5
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:2541]: Low this to 4 (Update jira ticket too)

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 4]
```diff
@@ -2360,6 +2515,91 @@ definition {
...
+	test HeadlesFieldIsNotMandatoryInSidePanelOnCreationAction {
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:2542]: name of the test should be Headle**s**sFieldIsNotMandatoryInSidePanelOnCreationAction

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 5]
```diff
@@ -2631,6 +2871,76 @@ definition {
 		}
 	}
 
+	@description = "LPS-186875 Confirm that when selecting the Side Panel type option, other fields are visible"
+	@priority = 4
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:2875]: Low this to 3 (Update jira ticket too)

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 6]
```diff
@@ -2631,6 +2871,76 @@ definition {
...
+				locator1 = "DataSet#FILTER_SELECTION");
+
+			AssertElementPresent(
+				key_type = "Headless Action Key",
+				locator1 = "DataSet#FILTER_SELECTION");
+		}
+	}
+
+	@description = "LPS-186875 Confirm that creation action changes are not saved by clicking the cancel button"
+	@priority = 4
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:2908]: Low this to 3 (Update jira ticket too)

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 7]
```diff
@@ -2705,54 +3015,316 @@ definition {
...
+				key_type = "Headless Action Key",
+				locator1 = "DataSet#FILTER_SELECTION");
 		}
 	}
 
-	@description = "LPS-194351 Confirm that when editing an already created modal action, it is not possible to leave the fields empty in the Creation Action"
-	@priority = 4
-	test TheModalActionFieldsCannotBeEmptyInCreationAction {
+	@description = "LPS-186875 Confirm that the Headles action key field is not mandatory"
+	@priority = 5
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:3074]: Low this to 4 (Update jira ticket too)

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 8]
```diff
@@ -2705,54 +3015,316 @@ definition {
...
+	test TheHeadlesFieldIsNotMandatoryOnCreationAction {
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:3075]: Name should be Headles**s**FieldIsNotMandatoryOnCreationAction (Dont to use "The" on every test name)

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 9]
```diff
@@ -2705,54 +3015,316 @@ definition {
...
-				actionName = "Edit",
-				actionTab = "Creation Actions",
-				icon = "pencil",
-				readlessAction = "xx-ee",
-				type = "Link",
-				url = "https://gist.github.com/deborasoliveira/40716d0b03d33ff7e636ca5d366c79c5#file-link-action-url-example");
-		}
+	@description = "LPS-186875 Confirm that when clicking on the collapse buttons some fields are hidden"
+	@priority = 3
+	test TheCollapseButtonsHideFieldsOnCreationAction {
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:3020]: Remove "The" from the test names

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 10]
```diff
@@ -2705,54 +3015,316 @@ definition {
...
+
+			AssertTextEquals.assertPartialText(
+				locator1 = "Message#SUCCESS",
+				value1 = "Your request completed successfully.");
+		}
+	}
+
+	@description = "LPS-186875 Confirm that you can select an item in the item field"
+	@priority = 5
+	test TheIconCanBeSelectedOnCreationAction {
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:3131]: Remove "The" from the test names

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 11]
```diff
@@ -2705,54 +3015,316 @@ definition {
...
+			Button.clickSave();
+
+			AssertTextEquals.assertPartialText(
+				locator1 = "Message#SUCCESS",
+				value1 = "Your request completed successfully.");
+		}
+	}
+
+	@description = "LPS-186875 Confirm that you can select an item in the item field"
+	@priority = 5
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:3130]: Low this to 4 (Update jira ticket too)

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 12]
```diff
@@ -2705,54 +3015,316 @@ definition {
...
+			AssertElementPresent(
+				key_position = 1,
+				keyName = "Update",
+				locator1 = "DataSet#FIELDS_TABLE");
+		}
+	}
+
+	@description = "LPS-186875 Confirm that the "Icon" field is not mandatory"
+	@priority = 3
+	test TheIconFieldIsNotMandatoryOnCreationAction {
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:3187]: Remove "The" from the test names

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 13]
```diff
@@ -2705,54 +3015,316 @@ definition {
...
+			AssertElementPresent(
+				key_position = 1,
+				keyName = "Update",
+				locator1 = "DataSet#FIELDS_TABLE");
+		}
+	}
+
+	@description = "LPS-186875 Confirm that the "Label" is mandatory to create action"
+	@priority = 3
+	test TheLabelFieldIsMandatoryOnCreationAction {
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:3233]: Remove "The" from the test names

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 14]
```diff
@@ -2705,54 +3015,316 @@ definition {
...
+				locator1 = "Message#ERROR_FORM_FIELD_REQUIRED",
+				value1 = "This field is required.");
+
+			AssertElementPresent(locator1 = "DataSet#ANY_BUTTON_DISABLED");
+		}
+	}
+
+	@description = "LPS-194351 Confirm that when editing an already created Link action, it is not possible to leave the fields empty in the Creation Action"
+	@priority = 4
+	test TheLinkActionInCreationActionFieldsCannotBeEmpty {
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:3282]: Remove "The" from the test names

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 15]
```diff
@@ -2705,54 +3015,316 @@ definition {
...
+
+	@description = "LPS-186875 Confirm that the "Label" is mandatory to create action"
+	@priority = 3
+	test TheLabelFieldIsMandatoryOnCreationAction {
+		task ("And clicks on Create Item Action button") {
+			Button.click(button = "Creation Actions");
+
+			Click(
+				key_text = "New Creation Action",
+				locator1 = "CustomerPortalSite#BUTTON");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:3239]: Use generic button path

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 16]
```diff
@@ -2705,54 +3015,316 @@ definition {
...
+				locator1 = "Message#ERROR_FORM_FIELD_REQUIRED",
+				value1 = "This field is required.");
+
+			AssertElementPresent(locator1 = "DataSet#ANY_BUTTON_DISABLED");
+		}
+	}
+
+	@description = "LPS-194351 Confirm that when editing an already created modal action, it is not possible to leave the fields empty in the Creation Action"
+	@priority = 4
+	test TheModalActionFieldsCannotBeEmptyInCreationAction {
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:3327]: Remove "The" from the test names

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 17]
```diff
@@ -2801,4 +3373,203 @@ definition {
 		}
 	}
 
+	@description = "LPS-186875 Confirm that the title field is mandatory"
+	@priority = 3
+	test TheTitleFieldIsMandatoryOnCreationAction {
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:3378]: Remove "The" from the test names

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 18]
```diff
@@ -2801,4 +3373,203 @@ definition {
...
+		task ("And clicks on Create Item Action button") {
+			Button.click(button = "Creation Actions");
+
+			Click(
+				key_text = "New Creation Action",
+				locator1 = "CustomerPortalSite#BUTTON");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:3384]: Use generic button path

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 19]
```diff
@@ -2801,4 +3373,203 @@ definition {
...
+				locator1 = "Message#ERROR_FORM_FIELD_REQUIRED",
+				value1 = "This field is required.");
+
+			AssertElementPresent(locator1 = "DataSet#ANY_BUTTON_DISABLED");
+		}
+	}
+
+	@description = "LPS-186875 Confirm that the URL field is mandatory"
+	@priority = 4
+	test TheURLFieldIsMandatoryOnCreationAction {
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:3429]: Remove "The" from the test names

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 20]
```diff
@@ -2801,4 +3373,203 @@ definition {
...
+		task ("And clicks on Create Item Action button") {
+			Button.click(button = "Creation Actions");
+
+			Click(
+				key_text = "New Creation Action",
+				locator1 = "CustomerPortalSite#BUTTON");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:3435]: Use generic button path

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 21]
```diff
@@ -2801,4 +3373,203 @@ definition {
...
+		task ("And the user fills Title field and URL") {
+			Type(
+				key_fieldName = "add-here-the-title-of-the-modal",
+				locator1 = "DataSet#PRESELECT_VALUE_INPUT",
+				value1 = "Title");
+
+			Type(
+				key_fieldId = "URL",
+				locator1 = "TextArea#ANY_ID",
+				value1 = "https://gist.github.com/deborasoliveira/40716d0b03d33ff7e636ca5d366c79c5#file-link-action-url-example");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:3454]: What is this debora's sample?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 22]
```diff
@@ -2801,4 +3373,203 @@ definition {
...
+			AssertTextEquals(
+				locator1 = "Message#ERROR_FORM_FIELD_REQUIRED",
+				value1 = "This field is required.");
+
+			AssertElementPresent(locator1 = "DataSet#ANY_BUTTON_DISABLED");
+		}
+	}
+
+	@description = "LPS-186875 Confirm that the URL field is mandatory"
+	@priority = 4
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:3428]: Low this to 3 (Update jira ticket too)

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 23]
```diff
@@ -2801,4 +3373,203 @@ definition {
...
+			AssertTextEquals(
+				locator1 = "Message#ERROR_FORM_FIELD_REQUIRED",
+				value1 = "This field is required.");
+
+			AssertElementPresent(locator1 = "DataSet#ANY_BUTTON_DISABLED");
+		}
+	}
+
+	@description = "LPS-186875 Confirm that the URL field is mandatory in the Side Panel type"
+	@priority = 4
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:3525]: Low this to 3 (Update jira ticket too)

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 24]
```diff
@@ -2801,4 +3373,203 @@ definition {
...
+	test URLFieldIsMandatoryInSidePanelOnCreationAction {
+		task ("And clicks on Create Item Action button") {
+			Button.click(button = "Creation Actions");
+
+			Click(
+				key_text = "New Creation Action",
+				locator1 = "CustomerPortalSite#BUTTON");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:3532]: Use generic button path

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 25]
```diff
@@ -2801,4 +3373,203 @@ definition {
...
+
+	@description = "LPS-186875 Confirm that the Title field is mandatory in the Side Panel type"
+	@priority = 3
+	test TitleFieldIsMandatoryInSidePanelOnCreationAction {
+		task ("And clicks on Create Item Action button") {
+			Button.click(button = "Creation Actions");
+
+			Click(
+				key_text = "New Creation Action",
+				locator1 = "CustomerPortalSite#BUTTON");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:3481]: Use generic button path

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 26]
```diff
@@ -2705,54 +3015,316 @@ definition {
...
+
+	@description = "LPS-186875 Confirm that the "Icon" field is not mandatory"
+	@priority = 3
+	test TheIconFieldIsNotMandatoryOnCreationAction {
+		task ("And clicks on Create Item Action button") {
+			Button.click(button = "Creation Actions");
+
+			Click(
+				key_text = "New Creation Action",
+				locator1 = "CustomerPortalSite#BUTTON");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:3193]: Use generic button path

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 27]
```diff
@@ -2705,54 +3015,316 @@ definition {
...
+
+	@description = "LPS-186875 Confirm that you can select an item in the item field"
+	@priority = 5
+	test TheIconCanBeSelectedOnCreationAction {
+		task ("And clicks on Create Item Action button") {
+			Button.click(button = "Creation Actions");
+
+			Click(
+				key_text = "New Creation Action",
+				locator1 = "CustomerPortalSite#BUTTON");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:3137]: Use generic button path

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 28]
```diff
@@ -2705,54 +3015,316 @@ definition {
...
-	test TheModalActionFieldsCannotBeEmptyInCreationAction {
+	@description = "LPS-186875 Confirm that the Headles action key field is not mandatory"
+	@priority = 5
+	test TheHeadlesFieldIsNotMandatoryOnCreationAction {
+		task ("And clicks on Create Item Action button") {
+			Button.click(button = "Creation Actions");
+
+			Click(
+				key_text = "New Creation Action",
+				locator1 = "CustomerPortalSite#BUTTON");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:3081]: Use generic button path

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 29]
```diff
@@ -2705,54 +3015,316 @@ definition {
...
+		task ("And clicks on Create Item Action button") {
+			Button.click(button = "Creation Actions");
 
-		task ("When the user clicks on the new item action 3 dots button and edit") {
 			Click(
-				key_index = 2,
-				key_text = "Actions",
-				locator1 = "DataSet#VERTICAL_ELLIPSIS_N");
+				key_text = "New Creation Action",
+				locator1 = "CustomerPortalSite#BUTTON");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:3026]: Use generic button path

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 30]
```diff
@@ -2631,6 +2871,76 @@ definition {
...
+
+	@description = "LPS-186875 Confirm that creation action changes are not saved by clicking the cancel button"
+	@priority = 4
+	test NewFieldsInTheModalTypeOnCreationAction {
+		task ("And clicks on Create Item Action button") {
+			Button.click(button = "Creation Actions");
+
+			Click(
+				key_text = "New Creation Action",
+				locator1 = "CustomerPortalSite#BUTTON");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:2915]: Use generic button path

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 31]
```diff
@@ -2631,6 +2871,76 @@ definition {
...
 
+	@description = "LPS-186875 Confirm that when selecting the Side Panel type option, other fields are visible"
+	@priority = 4
+	test NewFieldsInSidePanelTypeOnCreationAction {
+		task ("And clicks on Create Item Action button") {
+			Button.click(button = "Creation Actions");
+
+			Click(
+				key_text = "New Creation Action",
+				locator1 = "CustomerPortalSite#BUTTON");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:2882]: Use generic button path

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 32]
```diff
@@ -2360,6 +2515,91 @@ definition {
...
+
+	@description = "LPS-186875 Confirm that an action creation can be saved with a different translation, and using the Side Panel type"
+	@priority = 5
+	test HeadlesFieldIsNotMandatoryInSidePanelOnCreationAction {
+		task ("And clicks on Create Item Action button") {
+			Button.click(button = "Creation Actions");
+
+			Click(
+				key_text = "New Creation Action",
+				locator1 = "CustomerPortalSite#BUTTON");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:2548]: Use generic button path

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 33]
```diff
@@ -1158,6 +1239,46 @@ definition {
...
+	@priority = 3
+	test CanDragAndDropOnCreationAction {
+		task ("Given two or more creation actions") {
+			DataSetAdmin.addActions(
+				actionName = "delete",
+				actionTab = "Creation Actions",
+				icon = "trash",
+				readlessAction = "xx-ee",
+				type = "Link",
+				url = "https://gist.github.com/deborasoliveira/40716d0b03d33ff7e636ca5d366c79c5#file-link-action-url-example");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:1252]: What is this deborasoliveira example?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 34]
```diff
@@ -1158,6 +1239,46 @@ definition {
...
+
+			DataSetAdmin.addActions(
+				actionName = "label",
+				actionTab = "Creation Actions",
+				icon = "pencil",
+				readlessAction = "product",
+				type = "Link",
+				url = "https://gist.github.com/deborasoliveira/40716d0b03d33ff7e636ca5d366c79c5#file-link-action-url-example");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:1260]: What is this deborasoliveira example?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 35]
```diff
@@ -491,6 +525,53 @@ definition {
...
 
+	@description = "LPS-186875 Confirm that creation action changes are not saved by clicking the cancel button"
+	@priority = 4
+	test CanCancelCreationActionOnCancelButton {
+		task ("And clicks on Create Item Action button") {
+			Button.click(button = "Creation Actions");
+
+			Click(
+				key_text = "New Creation Action",
+				locator1 = "CustomerPortalSite#BUTTON");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:536]: Use generic button path

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 36]
```diff
@@ -252,6 +252,40 @@ definition {
...
 
+	@description = "LPS-186875 Confirm that the Variant modal is displayed with the correct values"
+	@priority = 4
+	test CanAssertVariantModalIsPresentOnCreationAction {
+		task ("And clicks on Create Item Action button") {
+			Button.click(button = "Creation Actions");
+
+			Click(
+				key_text = "New Creation Action",
+				locator1 = "CustomerPortalSite#BUTTON");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:263]: Use generic button path

**Review by CarlosBrichete — CHANGES_REQUESTED** [review]
Hey @andreluis2022, please check my change requests

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/DataSet.path [hunk 37]
```diff
@@ -514,6 +514,31 @@
...
+<tr>
+	<td>SELECT_DRAG_OPTION</td>
+	<td>//div[@aria-labelledby="New Creation Action"]//table[contains(@class,'table-list')]//tr[contains(@class,'orderable-table-row') and contains(.,'${key_nameField}')]//td[@class="drag-handle-cell"]//button[contains(@aria-label, 'Drag')]</td>
+	<td></td>
+</tr>
+<tr>
+	<td>SELECT_LABEL_ITEM</td>
+	<td>//div[@aria-labelledby="New Creation Action"]//table[contains(@class,'table-list')]//tr[@class='orderable-table-row']//span[contains(.,'${key_name}')]</td>
+	<td></td>
+</tr>
```
[the remark this answers is not in the document]
Comment (andreluis2022) [reply portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/DataSet.path:536]: Hey @CarlosBrichete, both paths that were created are necessary, as there are no specific paths to determine the differences between elements, which is why I added an excerpt `//div[@aria-labelledby="New Creation Action"]` . This way we can differentiate the specific elements of `Creation Actions`, what do you think?

**Comment by dsanz** [comment]
Already covered in playwright

**Pull request closed without merging**